pyquil/device.py

```diff
@@ -275,26 +275,31 @@ class Device(object):
     def __init__(self, name, raw):
         """
         :param name: name of the device
-        :param raw: raw JSON response from the server with additional information about this device
+        :param raw: raw JSON response from the server with additional information about this device.
+        Two additional keys are optionally expected in device._raw - 'isa' and 'noise_model' - which
+        should conform to the dictionary format required by the .from_dict() methods for ISA and
+        NoiseModel, respectively.
         """
         self.name = name
-        self.raw = raw
+        self._raw = raw
+        self.isa = ISA.from_dict(raw['isa']) if 'isa' in raw else None
```

Comment:
raw is of this form:
```
{u'is_online': False,
 u'is_retuning': False,
 u'isa': {},
 u'noise_model': {},
 u'specs': {}}
```
therefore this will throw an error if we havent put the ISA (or NoiseModel in the line below) in the DB